substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/snippets/ReflectionPlugins.java

```diff
@@ -167,6 +171,13 @@ public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Rec
             }
         });
 
+        r.register1("getDeclaredFields", Receiver.class, new InvocationPlugin() {
+            @Override
+            public boolean apply(GraphBuilderContext b, ResolvedJavaMethod targetMethod, Receiver receiver) {
+                return processGetFields(b, targetMethod, receiver, snippetReflection, true, analysis, hosted);
+            }
+        });
```

Comment:
`getFields()` could be intrinsified similarly. It looks like `processGetFields()` is ready to accept it.